advanced_dialog.cpp

```diff
@@ -544,47 +551,77 @@ void AdvancedDialog::SetPixelSize(double val)
         m_pCameraCtrlSet->SetPixelSize(val);
 }
 
-// Needed to handle reset if the image scale changes on the fly
-void AdvancedDialog::ResetGuidingParams()
+// Get the best estimate for the current mount guide speeds
+double AdvancedDialog::DetermineGuideSpeed()
 {
-    LoadValues();           // Insure that control values reflect actual device properties and any changes made outside the AD
-    m_pMountPane->ResetRAGuidingParams();
-    m_pMountPane->ResetDecGuidingParams();
-    UnloadValues();
-}
+    double const siderealSecsPerSec = 0.9973;
+    double sidRate = 0.5;
+    #define RateX(spd)  (spd * 3600.0 / (15.0 * siderealSecsPerSec))

```

Comment:
minor: better define this as a real function and not a macro. reason: type safety and macro operator precedence problems
```
inline static double SiderealRateFromGuideSpeed(double guideSpeed)
{
    double const siderealSecsPerSec = 0.9973;
    return guideSpeed * 3600.0 / (15.0 * siderealSecsPerSec);
}
```

Reply:
Makes sense, thanks.